### PR TITLE
feat(enrichment): [R10] P3 honest user-agent on all browser fetch surfaces

### PR DESCRIPTION
**Pull request by ebarti**
## Phase

**P3**, part 1 of browser routing — the **honest-UA** swap across every anonymous Playwright crawl path (surfaces **#4 smart-extract, #5 enrichment detail live batch, #6 detail port ref impl, #7 WTTJ**). Stacks on **P2d-jobspy** (#310). Base: `feat/r10-p2d-jobspy`.

## What

Rip-and-replace: deletes the spoofed `Mozilla/5.0 … AppleWebKit/537.36` desktop-browser UA from the browser paths; the single honest-UA policy sources every browser context's identity.

- `enrichment/detail.py` `UA`, `discovery/smartextract.py` `UA`, `infrastructure/enrichment/playwright_fetcher.py` `_USER_AGENT` all resolve to `default_honest_user_agent().header_value()` (`JobHunter/<version> (+repo)`). Covers `new_page`/`new_context` at detail, WTTJ, smart-extract, and the detail port fetcher.
- **Authenticated LinkedIn persistent context (#8)** — a real, user-logged-in Chrome session, semantically distinct from anonymous crawling — now presents **its own real browser identity** (`user_agent=None`) instead of a spoofed literal. Owner-scoped posture recorded per **D1/D3**: the product does not evade controls, but must not break a user-authorized flow.

## Scope note

Pre-navigation robots/budget gating and the `SITE_DELAYS` → shared-limiter swap (the parallel-mode pacing fix) land in **P3 part 2** (next stacked PR). This PR is the isolated, fully-verifiable honest-UA change — the single most-cited browser defect ("live crawlers spoof a desktop-browser UA").

## Enforcement proof

- `test_fetch_surface_enforcement.py`: no `Mozilla`/`AppleWebKit` literal remains in the browser surfaces, and the UA constants resolve to the honest identity.

## Validation

- `uv --project workers/automation run --extra dev pytest -q -p no:randomly` -> **1846 passed**
- `uv --project workers/automation run --extra dev ruff check .` -> **All checks passed!**

## Docs

Deferred to P5 (safety notes + the LinkedIn/UA posture + ADR).

### CODE REVIEW

File: workers/automation/src/jobhunter/enrichment/detail.py [hunk 1]
```diff
@@ -82,10 +82,15 @@
...
+from jobhunter.domain.ports.politeness import default_honest_user_agent
 
 log = logging.getLogger(__name__)
 
-UA = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
+# Honest outbound identity for anonymous crawl paths (R10). Never impersonate a
+# browser on a surface we control. The authenticated LinkedIn persistent context
+# is the owner's real Chrome session and presents its own browser identity
+# (user_agent=None) rather than this bot UA — an owner-scoped posture (D1/D3).
+UA = default_honest_user_agent().header_value()
```
Comment (ebarti) [inline workers/automation/src/jobhunter/enrichment/detail.py:93]: PR Reviewer - pr-reviewer

Severity: High
Gate: Fail
Evidence: The comment on this hunk (detail.py:89-92) states the authenticated LinkedIn persistent context "presents its own browser identity (user_agent=None) rather than this bot UA". But that context is constructed in TWO places and only one was updated. `_default_linkedin_apply_resolver_factory` (detail.py:1379) now passes `user_agent=None`, while `scrape_site_batch` (detail.py:690-694) still passes `user_agent=UA`. Both call `LinkedInApplyUrlResolver(...).start()` -> `launch_persistent_context(user_data_dir=default_linkedin_apply_profile_dir(), ...)` against the SAME logged-in profile (linkedin_apply_resolver.py:123, 144-176). In `start()`, `if self._user_agent:` (line 171) means `None` omits the key (real Chrome UA) while `UA` stamps "JobHunter/<version> (+github)". So the primary LinkedIn batch-enrichment path (the recommended path per playwright_fetcher.py:51-54) drives the authenticated, logged-in session presenting a self-identifying bot UA -- exactly the "must not break a user-authorized flow" outcome this PR's own rationale (D1/D3) chose user_agent=None to avoid. Pre-PR both sites shared the same (spoofed) UA and were consistent; this PR introduces the divergence, so surface #8 is only half-migrated and the code contradicts its own comment.
Reproduction: Reasoned from code (grep of both LinkedInApplyUrlResolver(...) construction sites + resolver.start(); no live LinkedIn fetch performed).

The authenticated-LinkedIn honest-identity decision is applied to only one of two identical construction sites.

Required fix: Make scrape_site_batch's authenticated resolver (detail.py:690-694) pass user_agent=None too, unifying surface #8 -- OR, if the two authenticated sites are meant to differ, say so explicitly in the comment/PR body. Either way add a regression test pinning the authenticated resolver's user_agent on both sites.

File: workers/automation/tests/test_fetch_surface_enforcement.py [hunk 2]
```diff
@@ -48,6 +48,33 @@ def test_routed_non_browser_surfaces_have_no_adhoc_urllib() -> None:
...
+
+
+def test_browser_surfaces_have_no_spoofed_browser_ua() -> None:
+    for rel in BROWSER_UA_SURFACES:
+        source = (SRC_ROOT / rel).read_text(encoding="utf-8")
+        assert "Mozilla" not in source, f"{rel} still contains a spoofed browser UA"
+        assert "AppleWebKit" not in source, f"{rel} still contains a spoofed browser UA"
+
+
+def test_browser_ua_constants_resolve_to_the_honest_identity() -> None:
```
Comment (ebarti) [inline workers/automation/tests/test_fetch_surface_enforcement.py:112]: PR Reviewer - pr-reviewer

Severity: Medium
Gate: Pass
Evidence: This file pins the three honest-UA constants (UA/_USER_AGENT start with "JobHunter/", no "Mozilla") but nothing asserts the second, safety-relevant half of the PR: that the authenticated LinkedIn persistent context is built with user_agent=None. With no coverage on that behavior, the batch-vs-factory divergence (detail.py:690-694 user_agent=UA vs detail.py:1379 user_agent=None) ships unnoticed. A focused test inspecting the resolver construction (e.g. asserting the factory-built resolver's user_agent is None so start() omits it and Chrome uses its real UA) would have caught it.
Reproduction: Reasoned from code (no test references user_agent=None or the resolver factory).

The changed authenticated-LinkedIn UA behavior has no regression coverage.

Required fix: Add a focused test asserting the authenticated LinkedIn resolver is constructed with user_agent=None, covering both construction sites.

**Review by ebarti — COMMENTED** [review]
PR Reviewer - pr-reviewer -- COMMENT review (no approve / no request-changes)

Scope reviewed: honest-UA swap on the anonymous browser crawl surfaces (#4 smart-extract, #5 detail batch, #6 detail port fetcher, #7 WTTJ) + authenticated LinkedIn context user_agent=None. Diff = 4 files, +49/-7.

Verdict: Gate FAIL -- 1 High, 1 Medium.

Correct:
- Spoof eradication: zero Mozilla/AppleWebKit literals remain anywhere in workers/automation/src (repo-wide grep). All four browser surfaces resolve UA to default_honest_user_agent().header_value() = "JobHunter/<version> (+https://github.com/ebarti/JobHunter)". Clean distinction: anonymous crawl paths explicitly stamp the honest bot UA on new_page/new_context; the authenticated LinkedIn factory uses user_agent=None so Chrome's native UA passes through (verified via resolver.start() line 171: `if self._user_agent:`).
- No new stealth: no viewport spoofing, navigator.webdriver hiding, AutomationControlled flag, or extra_http_headers added in the touched files or the resolver.
- Apply safety seam untouched: git diff --name-only shows nothing under apply/ and chrome.py unchanged.
- Deferrals stated: robots/rate gating -> P3(part 2); docs -> P5.
- Verification: enforcement test 4 passed; full suite 1846 passed; ruff `All checks passed!`.

Blocking (High, inline on detail.py): the authenticated LinkedIn persistent context (surface #8) is built in two places; only the factory got user_agent=None, while scrape_site_batch still passes user_agent=UA on the same logged-in profile -- contradicting the PR's own comment/rationale and risking the user-authorized LinkedIn flow.

Non-blocking: Medium (inline on the test) -- no regression test for the user_agent=None authenticated behavior. Low -- an outbound-identity/safety change ships now with all docs deferred to P5; acceptable if P5 lands the safety note + LinkedIn/UA posture.

File: workers/automation/src/jobhunter/enrichment/detail.py (hunk 1 above)
Comment (ebarti) [reply workers/automation/src/jobhunter/enrichment/detail.py:93]: Fixed in b4516e26. The `scrape_site_batch` inline construction now passes `user_agent=None`, unifying surface #8 with the factory so the authenticated LinkedIn context presents the owner's real browser identity on the primary batch path (D1/D3), matching the module comment.

Added the both-sites test that would have caught this divergence: `test_authenticated_linkedin_context_never_uses_bot_ua_at_any_site` enumerates every `LinkedInApplyUrlResolver(...)` construction in detail.py via AST and asserts `user_agent=None` at each site, plus `test_linkedin_apply_resolver_factory_uses_owner_browser_identity` checks the factory attribute directly. Verified the AST test fails on the pre-fix source (finds `Name(id='UA')` at the batch site) and passes after. `pytest -k 'detail or enrichment or fetch_surface or linkedin'` = 154 passed; ruff clean.

Comment (ebarti) [reply workers/automation/src/jobhunter/enrichment/detail.py:93]: PR Reviewer - pr-reviewer (re-check from PR #314 review)

Resolution confirmed: this High is fixed by b4516e26 (merged forward into #314 via cc52ed32).

- detail.py now constructs LinkedInApplyUrlResolver(..., user_agent=None) at BOTH sites: the module factory `_default_linkedin_apply_resolver_factory` (detail.py:1622) and the `scrape_site_batch` inline construction (detail.py:901). The owner's authenticated session no longer self-identifies as the JobHunter/<ver> bot UA on the primary batch path.
- Regression guard is real and bites: `test_authenticated_linkedin_context_never_uses_bot_ua_at_any_site` (test_fetch_surface_enforcement.py) AST-enumerates every `LinkedInApplyUrlResolver(...)` construction in detail.py, requires >= 2 sites, and asserts each passes `user_agent=None` — it would have caught the original factory-vs-batch divergence. `test_linkedin_apply_resolver_factory_uses_owner_browser_identity` also checks the factory instance directly.

This High can be marked resolved. It flips #312's UA-unification gate.
